learn/core_concepts/primary_key.md

```diff
@@ -141,40 +141,40 @@ The code below updates the primary key to `title`:
 
-If Meilisearch cannot find a suitable attribute, the document addition process will be interrupted and no documents will be added to your index.
+If Meilisearch finds multiple attributes ending with `id`, it will throw the [`index_primary_key_multiple_candidates_found`](#index-primary-key-multiple-candidates-found) error and ask you to manually set the primary key. If it cannot find a suitable attribute, it will throw the [`index_primary_key_no_candidate_found`](#index-primary-key-no-candidate-found) error. In both cases, the document addition process will be interrupted and no documents will be added to your index.
 
 ## Primary key errors
 
 This section covers some primary key errors and how to resolve them.
 
-### `primary_key_inference_failed`
+### `index_primary_key_no_candidate_found`
 
-This error occurs when you add documents for the first time and Meilisearch [fails to guess your primary key](#meilisearch-guesses-your-primary-key). It can be resolved by [manually setting the index's primary key](#setting-the-primary-key-on-document-addition), or ensuring that all documents you add possess an `id` attribute.
+This error occurs when you add documents for the first time and none of them have an attribute ending with `id`. It can be resolved by [manually setting the index's primary key](#setting-the-primary-key-on-document-addition), or ensuring that all documents you add possess an `id` attribute.
 
 ```json
 {
-    "uid":1,
-    "indexUid": "books",
-    "status": "failed",
-    "type": "documentAdditionOrUpdate",
-    "canceledBy": null,
-    "details":{
-        "receivedDocuments":5,
-        "indexedDocuments":null
-        },
-    "error":{
-        "message": "The primary key inference process failed because the engine did not find any fields containing `id` substring in their name. If your document identifier does not contain any `id` substring, you can set the primary key of the index.",
-        "code": "primary_key_inference_failed",
-        "type": "invalid_request",
-        "link": "https://docs.meilisearch.com/errors#primary_key_inference_failed"
-        },
-    "duration": "PT0.007479S",
-    "enqueuedAt": "2021-12-30T11:17:49.708824Z",
-    "startedAt": "2021-12-30T11:17:49.709934Z",
-    "finishedAt": "2021-12-30T11:17:49.716303Z"
-}
+  "uid": 1,
+  "indexUid": "books",
+  "status": "failed",
+  "type": "documentAdditionOrUpdate",
+  "canceledBy": null,
+  "details":{
+    "receivedDocuments":5,
+    "indexedDocuments":5
+  },
+  "error":{
+    "message": "The primary key inference process failed because the engine did not find any field ending with `id` in its name. Please specify the primary key manually using the `primaryKey` query parameter.",
```

Comment:
Ah, sorry for not noticing this earlier, but it looks like the error message is not the latest version we decided upon together (the [specification PR](https://github.com/meilisearch/specifications/pull/208) is [up to date](https://github.com/meilisearch/specifications/pull/208#issuecomment-1372005125))

```suggestion
    "message": "The primary key inference failed as the engine did not find any field ending with `id` in its name. Please specify the primary key manually using the `primaryKey` query parameter.",
```